### PR TITLE
Remove state generator param from CiviformOidcLogoutActionBuilder

### PR DESCRIPTION
**Pull request by MichaelZetune**
### Description

Remove state generator param from `CiviformOidcLogoutActionBuilder`. We already have it in the `OidcConfiguration`.

### Issue(s) this completes

Progress towards #4280

### CODE REVIEW

File: server/app/auth/oidc/applicant/LoginGovClientProvider.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public OidcClient get() {
```
Comment (MichaelZetune) [inline server/app/auth/oidc/applicant/LoginGovClientProvider.java:99]: We set a state generator on line 86. This was originally duplicately set on 102.

Comment (avrittrohwer) [reply server/app/auth/oidc/applicant/LoginGovClientProvider.java:99]: I see that setStateGenerator() is called on the config, but this seems to be called on the client.  How does the config and the client interact?

Comment (avrittrohwer) [reply server/app/auth/oidc/applicant/LoginGovClientProvider.java:99]: Oh I see, the config is used in civiformOidcLogoutActionBuilder.java:106

Comment (MichaelZetune) [reply server/app/auth/oidc/applicant/LoginGovClientProvider.java:99]: `LoginGovClientProvider` indirectly extends `OidcClientProvider`, which has the base `OidcClient get()` method.

That base method calls `getConfig()` to get the `OidcConfiguration`, which should drop down to the lowest implementation available, here.

Let me know if you disagree.

**Review by avrittrohwer — APPROVED** [review]
Looks good to me, only request is to see if you can break any existing tests by intentionally messing this PR up and if not, add a test that breaks if not too much effort

**Pull request merged**